### PR TITLE
Make network errors retryable

### PR DESCRIPTION
**Pull request by simonz-bq**

### CODE REVIEW

File: src/errors/Errors.ts [hunk 1]
```diff
@@ -184,3 +184,17 @@ function isRetryableStatusCode(e: AWSError): boolean {
...
+ * Is the error caused by a network issue?
+ * @param e The client error caught.
+ * @returns True if the exception was caused by a network issue.
+ */
+function isNetworkingError(e: AWSError): boolean {
+    if (!e.originalError) {
+        return false;
+    } else {
+        const sourceError: AWSError = <AWSError> e.originalError;
+        return (sourceError.code == "NetworkingError");
```
Comment (guyilin-amazon) [inline src/errors/Errors.ts:198]: Better to use `===` instead of `==`

Comment (simonz-bq) [reply src/errors/Errors.ts]: Done

File: src/errors/Errors.ts [hunk 2]
```diff
@@ -162,14 +162,16 @@ export function isBadRequestException(e: AWSError): boolean {
 }
 
 /**
- * Is the exception a retryable exception?
+ * Is the exception a retryable exception given the state of the session's transaction?
  * @param e The client error caught.
```
Comment (byronlin13) [inline src/errors/Errors.ts:166]: Add onCommit docstring

Comment (simonz-bq) [reply src/errors/Errors.ts:166]: Done

**Review by kylepbit — DISMISSED** [review]

File: src/test/Errors.test.ts [hunk 3]
```diff
@@ -154,31 +154,57 @@ describe("Errors", () => {
...
         });
 
         it("should return true when error is NoHttpResponseException", () => {
             mockError.code = "NoHttpResponseException";
             mockError.statusCode = 200;
-            chai.assert.isTrue(isRetryableException(mockError));
+            chai.assert.isTrue(isRetryableException(mockError, false));
         });
 
         it("shoud return true when error is SocketTimeoutException", () => {
```
[the remark this answers is not in the document]
Comment (simonz-bq) [reply src/test/Errors.test.ts]: Done

**Review by kylepbit — DISMISSED** [review]

**Comment by justing-bq** [comment]
LGTM

**Review by kylepbit — APPROVED** [review]

**Pull request merged**